buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -807,6 +807,7 @@ public synchronized void stop(boolean tailLogs) {
         requireNonNull(esProcess, "Can't stop `" + this + "` as it was not started or already stopped.");
         // Test clusters are not reused, don't spend time on a graceful shutdown
         stopHandle(esProcess.toHandle(), true);
+        reaper.unregister(toString());
```

Comment:
Moving this is not directly related to the fix, but I think it was in the wrong place before because:

1. It was only called for forcible stops, not successful graceful stops
2. It was called multiple times with exactly the same arguments in the event of forcibly stopping a process tree with more than one process